app.js

```diff
@@ -0,0 +1,191 @@
+// 基礎編 問題
+
+// Q1 変数
+
+let nickname = 'ごっしー';
+let age = '28';
+let greet = '私のニックネームは' + nickname + 'です。年齢は' + age + 'です。';
+
+console.log(greet);
+
+
+// Q2 配列
+
+let languages = ['JavaScript', 'PHP', 'Ruby', 'Python', 'Go'];
+let templateText =`私の好きな言語は${languages[0]}です。次は${languages[3]}を勉強してみたいです。`;
+
+console.log(templateText);
+
+
+// Q3 オブジェクト
+
+let user = {
+  name: 'John',
+  age: 26,
+  bloodType: 'A',
+  favorite: 'card',
+};
+
+console.log(user.age)
+
+
+// Q4 配列×オブジェクト
+
+let playerList = [
+  {
+    name: 'John',
+    age: 26,
+    favorites: ['Card Game', 'Basket Ball', 'Programming'],
+  },
+  {
+    name: 'Bob',
+    age: 33,
+    favorites: ['Tinder', 'The Legend of Zelda'],
+  },
+  {
+    name: 'Michael',
+    age: 22,
+    favorites: ['Football', 'Smash Bros.'],
+  },
+];
+
+console.log(playerList[1].favorites[1]);
+
+
+// Q5 四則演算
+
+let johnAge = playerList[0].age;
+let bobAge = playerList[1].age;
+let michaelAge = playerList[2].age;
+
+let averageAge = (johnAge + bobAge + michaelAge) /playerList.length;
+
+
+console.log(averageAge);
+
+
+// Q6 関数
+
+function sayHello(){
+  console.log('Hello');
+}
+
+sayHello();
+
+let sayWorld = function() {
+  console.log('World');
+}
+
+sayWorld();
+
+
+// Q7 メソッド
+
+user.birthday = '2000-09-27';
+user.greet = function sayHello(){
+  console.log('Hello !');
+}
+
+user.greet();
+
+
+// Q8 引数
+
+let calc = {};
+
+calc.add = function(x, y) {
+  console.log(x + y);
+}
+
+calc.add(3, 4);
+
+
+calc.subtract = function(x, y) {
+  console.log(x - y);
+}
+
+calc.subtract(15, 5);
+
+
+calc.multiply = function(x, y) {
+  console.log(x * y);
+}
+
+calc.multiply(7, 7);
+
+
+calc.divide = function(x, y) {
+  console.log(x / y);
+}
+
+calc.divide(15, 3);
+
+
+// Q9 返り値
+
+function remainder(x, y) {
+  let divide = x % y;
+  return console.log(x + 'を' + y + 'で割った余りは' + divide + 'です。');
+}
+
+remainder(5, 3);
+
+
+// Q10 スコープ
+
+// 関数内で定義した変数のスコープは関数の範囲内なので、関数の外からは関数内で定義した変数Xは参照できないためエラーが発生した。
+
+
+
+// 応用編 問題
+
+// Q1 標準組み込みオブジェクト
+
+let random = Math.floor(Math.random()*8)+1;
+console.log(random);
+
+
+// Q2 コールバック関数
+
+setTimeout(function(){
+  console.log('Hello World!');
+}, 3000);
+
+
+// Q3 if
+
+let num = 0;
+
+if(num > 0) {
+  console.log('num is greater than 0');
+} else if(num < 0) {
+  console.log('num is less than 0');
+} else if(num === 0) {
+  console.log('num is 0');
+}
+
+
+// Q4 for
+
+let numbers = [];
+
+for(let i = 0; i<=99; i++) {
+  numbers[i] = i;
+}
+
+console.log(numbers);
+
+
+// Q5 for×if
+
+let mixed = [4, '2', 5, '8', '9', 0, 1];
+
+for(let i = 0; i < mixed.length; i++) {
+  if(typeof mixed[i] === 'number' && mixed[i] % 2 ===0) {
+    console.log('even');
+  } else if (typeof mixed[i] === 'number' && mixed[i] % 2 !== 0) {
+    console.log('odd');
+  } else if (typeof mixed[i] !== 'number') {
+    console.log('not number');
+  }

```

Comment:
[imo]
数値以外を弾くのであれば上記の条件で十分なので、elseの方が可読性・保守性共に向上します。

```suggestion
  } else {
    console.log('not number');
  }

```